File: 1754_coin_piles.hs
isValidPile :: [Int] -> Bool
isValidPile [x,y] =
    let 
        a = (2 * y - x) `div` 3
        b = (2 * x - y) `div` 3
    in a >=0 && b >= 0 && (2 * y - x) `mod` 3 == 0 && (2 * x - y) `mod` 3 == 0

yesOrNo :: Bool -> String
yesOrNo True = "YES"
yesOrNo False = "NO"

main :: IO()
main = do
    input <- getContents
    let (_:rest) = lines input
        piles = map (map (read :: String -> Int) . words) rest 
    let result = map (yesOrNo . isValidPile) piles
    mapM_ putStrLn result 

File: 1754_coin_piles.js
const fs = require("fs");
let input = fs.readFileSync(0, "utf-8").trim();
const [_, ...rest] = input.split("\n");
const piles = rest.map((x) => x.split(" ").map(Number));

function validPile(pile) {
    let [x, y] = pile;
    let a = (2 * y - x) / 3;
    let b = (2 * x - y) / 3;
    return Number.isInteger(a) && Number.isInteger(b) && a >= 0 && b >= 0;
}

let result = piles.map((pile) => (validPile(pile) ? "YES" : "NO")).join("\n");
console.log(result);
